src/libaktualizr/primary/download_nonostree_test.cc

```diff
@@ -0,0 +1,109 @@
+#include <gtest/gtest.h>
+
+#include <boost/process.hpp>
+
+#include "uptane_test_common.h"
+
+#include "config/config.h"
+#include "logging/logging.h"
+#include "package_manager/ostreemanager.h"
+#include "primary/aktualizr.h"
+#include "storage/sqlstorage.h"
+#include "test_utils.h"
+
+static std::string server = "http://127.0.0.1:";
+static std::string treehub_server = "http://127.0.0.1:";
+static boost::filesystem::path sysroot;
+
+static std::string new_rev;
+
+/*
+ * Reject non-OSTree binaries.
+ */
+TEST(Aktualizr, DownloadNonOstreeBin) {
+  TemporaryDirectory temp_dir;
+  Config conf = UptaneTestCommon::makeTestConfig(temp_dir, server);
+  conf.pacman.type = PackageManager::kOstree;
+  conf.pacman.sysroot = sysroot.string();
+  conf.pacman.ostree_server = treehub_server;
+  conf.pacman.os = "dummy-os";
+  conf.provision.device_id = "device_id";
+  conf.provision.ecu_registration_endpoint = server + "/director/ecus";
+  conf.tls.server = server;
+
+  {
+    std::shared_ptr<INvStorage> storage = INvStorage::newStorage(conf.storage);
+    auto uptane_client = SotaUptaneClient::newDefaultClient(conf, storage);
+    uptane_client->initialize();
+    EXPECT_FALSE(uptane_client->uptaneIteration());
+    EXPECT_STREQ(uptane_client->getLastException().what(),
+                 "The target had a non-OSTree package that can not be installed on an OSTree system.");
+  }
+}
+
+#ifndef __NO_MAIN__
+int main(int argc, char **argv) {
+  ::testing::InitGoogleTest(&argc, argv);
+
+  logger_init();
+
+  if (argc != 3) {
+    std::cerr << "Error: " << argv[0] << " requires the path to the uptane-generator utility "
+              << "and an OStree sysroot\n";
+    return EXIT_FAILURE;
+  }
+
+  boost::filesystem::path uptane_generator_path;
+  uptane_generator_path = argv[1];
+
+  TemporaryDirectory meta_dir;
+  TemporaryDirectory temp_sysroot;
+  sysroot = temp_sysroot / "sysroot";
+
+  Process ostree("ostree");
+
+  // uses cp, as boost doesn't like to copy bad symlinks
+  int ret = system((std::string("cp -r ") + argv[2] + std::string(" ") + sysroot.string()).c_str());
+  if (ret != 0) {
+    return -1;
+  }
+  auto r = ostree.run(
+      {"rev-parse", std::string("--repo"), (sysroot / "/ostree/repo").string(), "generate-remote/generated"});
+  if (std::get<0>(r) != 0) {
+    return -1;
+  }
```

Comment:
Is this `ostree` process stuff still necessary? Seems like that's a relic of the `aktualizr_fullostree_test.cc` style.